test/server/server_test.go

```diff
@@ -0,0 +1,331 @@
+//go:build integration
+// +build integration
+
+/*
+Copyright 2021 Andy Dalton
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package server_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/adalton/teleport-exercise/pkg/command"
+	"github.com/adalton/teleport-exercise/service/jobmanager/jobmanagerv1"
+	"github.com/adalton/teleport-exercise/util/grpcutil"
+	"google.golang.org/grpc"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Note: There is some fragility with these tests.
+//       * This expects port 12345 to be available (so we wouldn't want to
+//         run multiple instances of this test on the same host in parallel)
+//       * There's a Sleep() between starting the server and trying to connect
+//         to it.  That's open to race conditions.  I'd like to find a way to
+//         know that the server is up before we try to connect to it.
```

Comment:
Perhaps you could pass a `net.Listener` into the `RunJobmanagerServer` function. Then in each test you could listen on `:0`. To remove the `Sleep` you could attempt to dial the address of the `net.Listner` until you get a successful connection or until some defined timeout/number of tries has been reached.

Reply:
Thank you for the suggestions.  I'll take a look and try to make this more elegant.

Reply:
You could also potentially utilize a channel instead or repeatedly attempting to dial

Reply:
Added in other comment, but as long as you dial to a port that is open you should not have server/client startup issues - grpc conns are lazy and tend to do the right thing.